misc/python/materialize/mzcompose/composition.py

```diff
@@ -687,7 +678,6 @@ def up(
             for service in self.compose["services"].values():
                 service["entrypoint"] = ["sleep", "infinity"]
                 service["command"] = []
-            self._write_compose()
```

Comment:
What makes all of those `_write_compose()` calls redundant?

Reply:
We now write the compose file immediately before reading it.

The previous approach was to write it whenever we changed it.